pkg/backend/httpstate/backend.go

```diff
@@ -712,6 +712,29 @@ func (b *cloudBackend) GetStack(ctx context.Context, stackRef backend.StackRefer
 	return newStack(stack, b), nil
 }
 
+// Confirm the specified stack's project doesn't contradict the Pulumi.yaml of the current project.
+// if the CWD is not in a Pulumi project,
+//     does not contradict
+// if the project name in Pulumi.yaml is "foo".
+//     a stack with a name of foo/bar/foo should not work.
+func currentProjectContradictsWorkspace(stack client.StackIdentifier) bool {
+	projPath, err := workspace.DetectProjectPath()
+	if err != nil {
+		return false
+	}
+
+	if projPath == "" {
+		return false
+	}
+
+	proj, err := workspace.LoadProject(projPath)
```

Comment:
Is there a reason we can't use `b.currentProject` of type `*workspace.Project` and we need to re-detect and load it again from ambient information?